session.cpp

```diff
@@ -54,54 +60,155 @@ int session::handle_request(const boost::system::error_code& error,
 
 }
 
+// setup static file serving response
+int session::handle_static(std::string& url, http_response& response) {
+    std::cout << "========= Handling Static =========" << std::endl;
+    
+    std::string abs_path = get_exec_path() + "/public" + get_path_from_url(url);
+    std::cout << "Serving file from: " << abs_path << std::endl;
+    
+    if (!file_exists(abs_path)) {
+      // TODO: 404 Error
+      std::cerr << "Error: " << abs_path << " does not exist" << std::endl;
+      return -1;
+    } 
+
```

Comment:
"If a file isn’t present, you should return a 404 response code." this is from the spec, so we might want to have a an error request handler that can send back an error to the browser.